packages/block-editor/src/components/block-list/block-mobile-toolbar.native.js

```diff
@@ -18,17 +18,30 @@ import styles from './block-mobile-toolbar.scss';
 import BlockMover from '../block-mover';
 import InspectorControls from '../inspector-controls';
 
+const showBottomSheet = ( { showSettings, onSettingsToggle: onClose, ...rest } ) => (
+	<BottomSheet
+		isVisible={ showSettings }
+		onClose={ onClose }
+		hideHeader
+		{ ...rest }
+	>
+		<InspectorControls.Slot />
+	</BottomSheet>
+);
+
 const BlockMobileToolbar = ( {
 	clientId,
 	onDelete,
 	order,
+	...rest
 } ) => (
 	<View style={ styles.toolbar }>
 		<BlockMover clientIds={ [ clientId ] } />
 
 		<View style={ styles.spacer } />
 
-		<InspectorControls.Slot />
+		{ showBottomSheet( rest ) }
+		<InspectorControls.SlotButtons />
```

Comment:
On the second thought, in case we move the settings button here, we won't have anything left to fill this slot. So I'd be OK if we remove InspectorControls.SlotButtons altogether. We can add a new slot here in the future when needed.

Reply:
Another approach is to add prop to `InspectorControls` (e.g `buttons`) to define that this is a slot for buttons:

`<InspectorControls.Slot buttons />`

Then passing info about children in `InspectorControls` Fills to conditionally show/hide `Settings` button should be quite easy 

I will prepare some code to present this solution to make more clear view on that